app/components/primer/base_component.rb

```diff
@@ -152,6 +152,33 @@ def initialize(tag:, classes: nil, **system_arguments)
 
       raise ArgumentError, "`class` is an invalid argument. Use `classes` instead." if system_arguments.key?(:class) && !Rails.env.production?
 
+      if (denylist = system_arguments[:system_arguments_denylist])
+        if Rails.application.config.primer_view_components.force_system_arguments && !ENV["PRIMER_WARNINGS_DISABLED"]
```

Comment:
I'd like to extract the config check to `Primer::Component`, since it could be reused in other components

```rb
def force_system_arguments?
  Rails.application.config.primer_view_components.force_system_arguments
end
```